ReviewChecklist.md

```diff
@@ -0,0 +1,156 @@
+# PR review checklist
+The Solidity compiler is a critical piece of infrastructure in the Ethereum ecosystem and for this
+reason our review process is quite strict.
+The list below is meant to reduce the workload on the core team by helping contributors self-identify
+common issues before they are inevitably pointed out in the review.
+It's also meant as a final checklist for reviewers to go through before approving a PR.
+
+## Before you submit a PR
+- [ ] **Is the issue ready to be worked on?**
+    - If the issue does not have a desirability label (`nice to have`, `should have`,
+        `must have eventually`, `must have`, `roadmap`) we still have not decided whether to implement it.
+    - If the issue `needs design`, we have not yet decided how it should be implemented.
+    - `good first issue candidate` means that the issue will potentially be a `good first issue`
+        eventually but at the moment it's not yet ready to be worked on.
+- [ ] **Is this a breaking change?** Breaking changes should be based on the `breaking` branch rather than on `develop`.
+- [ ] **Does the PR actually address the issue?**
+    - [ ] Mention the issue number in the PR description.
+        If the PR solves it completely, use the `Fixes #<issue number>` form so that Github will close the issue automatically
+    - [ ] Do not include issue number in the PR title, branch name or commit description.
+- [ ] When submitting a PR from a fork **create a branch and give it a good name.**
+    E.g. `fix-array-abi-encoding-bug`.
+    Do not submit PRs directly from the `develop` branch of a fork since it makes rebasing and fetching new changes harder.
+- [ ] **Does the PR depend on other PRs?**
+    - [ ] If the PR has dependencies, mention them in bold in the description.
+    - [ ] Avoid basing PRs from forks on branches other than `develop` or `breaking` because
+        unfortunately gitub closes them when the base branch gets merged.
+        Do this only for PRs created directly in the main repo.
+
+## Coding style and good practices
+- [ ] Does the PR follow our [coding style](CODING_STYLE.md)?
+
+### Reliability
+- [ ] **Use assertions liberally.** If you are certain your assumption will not be be broken, prove it with `solAssert()`.
```

Comment:
```suggestion
- [ ] **Use assertions liberally.** If your solution assumes that a certain assertion will not be be violated, add a safeguard with `solAssert()`.
```

Reply:
I actually like the original wording more :) I could go in depth on when assertions are appropriate but that would be long. I think the current wording conveys most of that in a tongue-in-cheek way :)